daemon/src/projection.rs

```diff
@@ -261,11 +265,19 @@ impl Cfd {
                 (CfdState::ContractSetup, vec![])
             }
             ContractSetupCompleted { dlc } => {
-                self.details.tx_url_list.push(TxUrl::new(
-                    dlc.lock.0.txid(),
-                    network,
-                    TxLabel::Lock,
-                ));
+                // Lock transaction has shared output, but it likely has change
+                // output too that can be highlighted
+                let tx_url = TxUrl::new(dlc.lock.0.txid(), network, TxLabel::Lock);
+                // Find the relevant vout based on the locked amount as we don't
+                // have script pubkey for it.
+                let tx_url =
+                    if let Ok(outpoint) = dlc.lock.0.outpoint_of_value(dlc.lock_amount_for(role)) {
```

Comment:
📣 I don't think this is quite right. If the goal is to only highlight the `dlc_output` you need to add up the `taker_lock_amount` and the `maker_lock_amount`. The sum of those values is what makes up the `dlc_output`.

Reply:
But thinking about it, we shouldn't be using amounts to identify outputs if we can help it. For some reason I thought you would use the amount to identify change outputs of the lock transaction. Those cannot be easily identified otherwise, because `bdk` chooses a change address automatically during contract setup.

To identify the `lock_output` you can use this public method from `maia`: https://github.com/comit-network/maia/blob/close-tx-fee-rate/src/protocol.rs#L251-L264. Sorry for the confusion. I didn't realise the goal was to highlight the `lock_output`.

Reply:
I don't think we need to go back to `maia` here, that `lock_desc` was readily available in the `Dlc`. Now, the question arises: should we do the same for commit transaction? I can see that we have a descriptor there too?